src/components/StakingVaults/AllEarnOpportunities.tsx

```diff
@@ -21,27 +26,60 @@ export const AllEarnOpportunities = () => {
   const history = useHistory()
   const location = useLocation()
   const foxyInvestorFeatureFlag = useAppSelector(state => selectFeatureFlag(state, 'FoxyInvestor'))
+  const cosmosInvestorFlag = useAppSelector(state => selectFeatureFlag(state, 'CosmosInvestor'))
   const {
     state: { isConnected },
     dispatch,
   } = useWallet()
   const sortedVaults = useSortedYearnVaults()
   const { opportunities } = useFoxyBalances()
+  const { activeStakingOpportunities, stakingOpportunities } = useCosmosStakingBalances({
```

Comment:
Not a blocker but rather a question, how are we going to iterate from this when adding more Cosmos SDK chains, or in case we need to support different IBC assets? Would we need to call this hook repetitively for each CAIP19?

Reply:
I see too choices, either we will call this hook for each assetId, or we will pass the list of assetId directly to this hook which will handle it. I think it really depends of the specificy of the assets we may add. For cosmos SDK chains I guess we will be very close to what we are currently doing there. For other IBC assets it is hard to project without more info.

Reply:
Other Cosmos SDK chains (e.g Osmosis) and IBC assets should be fairly similar since we won't handle the denom on the client, so I would suppose that they effectively are the same.